### ISSUE TITLE
Monitoring v2 - failed to install in harden clusters

### ISSUE THREAD

**Issue by jiaqiluo** [rancher/rancher#28536]
**What kind of request is this (question/bug/enhancement/feature request):**
bug

**Steps to reproduce (least amount of steps as possible):**
- run the latest rancher:master-head in HA 
- provision a hardening custom cluster https://rancher.com/docs/rancher/v2.x/en/security/hardening-2.4/?query=harden
- install monitoring v2 into the cluster 

**Result:**
<img width="1409" alt="Screen Shot 2020-08-24 at 11 26 53 AM" src="https://user-images.githubusercontent.com/6218999/91085351-2d25da80-e602-11ea-982f-f400a6f0ca64.png">

- all PushProxy related deployments fail because of the following error
```
pods xxx is forbidden: unable to validate against any pod security policy: []:Deployment does not have minimum availability.
```

- pod of `rancher-monitoring-grafana` has a container with an incomplete status
```
NAME                                          READY   STATUS         RESTARTS   AGE
rancher-monitoring-grafana-795d4547fd-78mml   0/2     Init:Blocked   0          6m51s
```
<img width="1400" alt="Screen Shot 2020-08-24 at 12 02 44 PM" src="https://user-images.githubusercontent.com/6218999/91085357-30b96180-e602-11ea-8996-8cebe931ab9d.png">

- pod of `rancher-monitoring-prometheus-adapter` is unable to validate against any pod security policy
<img width="1405" alt="Screen Shot 2020-08-24 at 12 02 28 PM" src="https://user-images.githubusercontent.com/6218999/91085340-28612680-e602-11ea-9486-a929b4ebddf9.png">

gzrancher/rancher#12824

**Comment by jiaqiluo** [comment]
harden cluster in rancher:`master-f1a410a97e2abf8c81c62195a7b40f10007398f0-head`  docker install setup


![screencapture-206-189-215-122-p-c-tzhsq-p-cqnkg-workloads-2020-09-22-10_37_43](https://user-images.githubusercontent.com/6218999/93917988-927ff080-fcc0-11ea-8dde-cd4f17ceff0e.png)

The following error on the deployment `rancher-monitoring-grafana`

`Pods "rancher-monitoring-grafana-dbd7ccd7b-nbvfz" is forbidden: PodSecurityPolicy: unable to validate pod: [pod.metadata.annotations[container.seccomp.security.alpha.kubernetes.io/grafana-sc-datasources]: Forbidden: runtime/default is not an allowed seccomp profile. Valid values are docker/default pod.metadata.annotations[container.seccomp.security.alpha.kubernetes.io/grafana-sc-dashboard]: Forbidden: runtime/default is not an allowed seccomp profile. Valid values are docker/default pod.metadata.annotations[container.seccomp.security.alpha.kubernetes.io/grafana]: Forbidden: runtime/default is not an allowed seccomp profile. Valid values are docker/default pod.metadata.annotations[seccomp.security.alpha.kubernetes.io/pod]: Forbidden: seccomp may not be set pod.metadata.annotations[container.seccomp.security.alpha.kubernetes.io/grafana-sc-datasources]: Forbidden: seccomp may not be set pod.metadata.annotations[container.seccomp.security.alpha.kubernetes.io/grafana-sc-dashboard]: Forbidden: seccomp may not be set pod.metadata.annotations[container.seccomp.security.alpha.kubernetes.io/grafana]: Forbidden: seccomp may not be set]; Deployment does not have minimum availability.`

**Comment by jiaqiluo** [comment]
reopen this issue, see https://github.com/rancher/rancher/issues/29059#issuecomment-707997762 for the detailed results.

**Comment by jiaqiluo** [comment]
`v2.5-b72284720041c18fa2e64f63896c0388133aa9f6-head` 
still hit the same error https://github.com/rancher/rancher/issues/29059#issuecomment-707997762

**Comment by jiaqiluo** [comment]
The bug fix is validated on monitoring v2 chart `9.4.201` on 
- rancher:`v2.5-7eb1792946639251f51bd109b630f9c1cf8940c2-head`
- rancher:`master-3d755ff62ae975044b19d1b74c564e9920127f73-head`


Results:
- the monitoring v2 is installed successfully on a harden cluster 
- all endpoints (node-exporter, etcd, scheduler and controller-manager endpoints) are healthy in Prometheus 
- metrics show data in Grafana
